composer/callbacks/system_metrics_monitor.py

```diff
@@ -83,3 +134,17 @@ def compute_system_metrics(self):
         for k, v in network_usage.items():
             system_metrics[f'network_{k}'] = v
         return system_metrics
+
+    def compute_gpu_min_max_metrics(self, all_metrics, state):
```

Comment:
you could just pass `state.device` here instead of all of `state`. not strongly opinionated